tests/test_model_serializer.py

```diff
@@ -1333,3 +1338,60 @@ class Meta:
         serializer.save()
 
         self.assertEqual(instance.char_field, 'value changed by signal')
+
+
+class Issue2648Test(TestCase):
+    def test_model_serializer_uses_partial_update_extra_fields_when_not_empty(self):
```

Comment:
It would be nice to also have a test in which provided partial data only contains the extra fields to assure that it does not break.